### PR TITLE
[ besu ]Add code to compile and deploy smartcontracts

### PR DESCRIPTION
**Pull request by lakshyakumar**
Signed-off-by: lakshyakumar <lakshya.kumar@accenture.com>

**Changelog**
- Add code to compile and deploy smartcontracts

 

**Reviewed by**
@jagpreetsinghsasan 

 

**Linked issue**
#895

### CODE REVIEW

File: examples/supplychain-app/besu/smartContracts/compile.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                  '*': [ '*' ]
+              }
+          }
+      }
+    };
+
+  var output = JSON.parse(solc.compile(JSON.stringify(input))); // compiling the smart contract using the main entrypoint file
+  var smartContract = {}; // Creating an object to return only the required data to the caller
+  smartContract.bytecode = output.contracts[`${entrypoint}`][`${contractName}`].evm.bytecode.object; // adding bytecode to the variable
+  smartContract.abi = output.contracts[`${entrypoint}`][`${contractName}`].abi; // adding abi to the smartCOntract variable
```
Comment (jagpreetsinghsasan) [inline examples/supplychain-app/besu/smartContracts/compile.js:55]: spelling of smartContract

File: examples/supplychain-app/besu/smartContracts/deploy.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+const orionPublicKey = args['orionKey'];
+const privateKey = args['privateKey'];
+const privateFor = [];
+const outputFolder = args['output'] == true? args['output'] : './build';
+args['privateFor'].split(',').forEach(item=>privateFor.push(item));
+const numberOfIterations = args['numberOfIteration'] | 100;
+
+args['v'] && console.log(`Creating a web3 provider.......`);
+const web3 = new EEAClient(new Web3(`${url}`), `${chainId}`);// Creating a provider
+var transactionHash = "";  // to store transaction hash to get the transaction recipt
```
Comment (jagpreetsinghsasan) [inline examples/supplychain-app/besu/smartContracts/deploy.js:22]: Spelling of receipt

File: examples/supplychain-app/besu/smartContracts/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,3083 @@
...
+      "version": "2.10.0",
+      "resolved": "https://registry.npmjs.org/yauzl/-/yauzl-2.10.0.tgz",
+      "integrity": "sha1-x+sXyT4RLLEIb6bY5R+wZnt5pfk=",
+      "requires": {
+        "buffer-crc32": "~0.2.3",
+        "fd-slicer": "~1.1.0"
+      }
+    }
+  }
+}
```
Comment (jagpreetsinghsasan) [inline examples/supplychain-app/besu/smartContracts/package-lock.json:3083]: do we need this file?

Comment (lakshyakumar) [reply examples/supplychain-app/besu/smartContracts/package-lock.json:3083]: It describes the exact tree of dependencies generated by the npm and the same exact tree is used for the installation everytime.

File: examples/supplychain-app/besu/smartContracts/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Deploying Smart Contract on Besu network using javaScript
+
+## Requirments 
+- node<br>
+- npm<br>
+
+
+1. In terminal, change directory to besu/smartContracts and run command below to install the required packages listed in package.json .<br>
```
Comment (sownak) [inline examples/supplychain-app/besu/smartContracts/README.md:8]: give complete relative path

File: examples/supplychain-app/besu/smartContracts/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "smartcontracts",
+  "version": "1.0.0",
+  "description": "Ganache CLI<br>\r Truffle<br>",
+  "main": "compile.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (sownak) [inline examples/supplychain-app/besu/smartContracts/package.json:10]: if there are no tests, remove this

File: examples/supplychain-app/besu/smartContracts/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "smartcontracts",
+  "version": "1.0.0",
+  "description": "Ganache CLI<br>\r Truffle<br>",
+  "main": "compile.js",
+  "directories": {
+    "test": "test"
```
Comment (sownak) [inline examples/supplychain-app/besu/smartContracts/package.json:7]: and this as well

**Pull request merged**